File: ios/PearCal/LinkModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PearCalLink, NSObject)
RCT_EXTERN_METHOD(getPendingLink:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getPendingTab:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)
@end

File: ios/PearCal/LinkModule.swift
import Foundation

@objc(PearCalLink)
class LinkModule: NSObject {
  static var pendingLink: String? = nil
  static var pendingTab: String? = nil
  static var pendingGroupSettingsId: String? = nil

  @objc func getPendingLink(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject: RCTPromiseRejectBlock
  ) {
    let link = LinkModule.pendingLink
    LinkModule.pendingLink = nil
    resolve(link)
  }

  @objc func getPendingTab(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject: RCTPromiseRejectBlock
  ) {
    let tab = LinkModule.pendingTab
    LinkModule.pendingTab = nil
    resolve(tab)
  }

  @objc func getPendingGroupSettingsId(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject: RCTPromiseRejectBlock
  ) {
    let gsid = LinkModule.pendingGroupSettingsId
    LinkModule.pendingGroupSettingsId = nil
    resolve(gsid)
  }

  @objc static func requiresMainQueueSetup() -> Bool { return false }
}
